streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1435,6 +1439,13 @@ public Source(final String name,
                       final Set<String> topics,
                       final Pattern pattern) {
             super(name);
+            if (topics == null && pattern == null) {
+                throw new IllegalArgumentException("Either topics or pattern must be not-null, but both are null.");
+            }
+            if (topics != null && pattern != null) {
+                throw new IllegalArgumentException("Either topics or pattern must be null, but both are not null.");
+            }
+
```

Comment:
What should happen if the topic or the pattern is empty?

Reply:
I don't think those are special cases. For `Pattern.compile(null)` it throws NPE, so we would never hit this code but fail before.

`Pattern.compile("")` should be fine but just don't match any topic. Similar, to empty topic list. For both cases, the consumer should just be idle.

\cc @guozhangwang or @hachikuji to confirm.